smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -280,11 +343,22 @@ def _on_iteration(self) -> None:
 
             response = build_reply(reply)
 
+        timings.append(time.perf_counter() - interm)
+        interm = time.perf_counter()
+
         # serialized = self._worker.serialize_reply(request, transformed_output)
         serialized_resp = MessageHandler.serialize_response(response)  # type: ignore
+
+        timings.append(time.perf_counter() - interm)
+        interm = time.perf_counter()
         if request.callback:
             request.callback.send(serialized_resp)
 
+        timings.append(time.perf_counter() - interm)
+        interm = time.perf_counter()
+
+        print(" ".join(str(time) for time in timings))
```

Comment:
consider adding a custom log level e.g. "perf" so you can configure your logging output w/familiar logging interface / avoid prints

Reply:
That's an interesting suggestion, thanks